clients/java/client/src/main/java/org/camunda/bpm/client/ExternalTaskClientBuilder.java

```diff
@@ -33,12 +32,42 @@ public interface ExternalTaskClientBuilder {
 
   /**
    * Base url of the Camunda BPM Platform REST API. This information is mandatory.
+   * <p>
+   * If this method is used, it will create a permanent url resolver with the given baseUrl.
    *
    * @param baseUrl of the Camunda BPM Platform REST API
    * @return the builder
    */
   ExternalTaskClientBuilder baseUrl(String baseUrl);
 
+  /**
+   * Url resolver of the Camunda BPM Platform REST API. This information is mandatory.
+   * <p>
+   * If the server is in a cluster or you are using spring cloud, you can create a class which implements UrlResolver..
+   * <p>
+   * this is a sample for spring cloud DiscoveryClient
+   * <p>
+   * public class CustomUrlResolver implements UrlResolver{
+   * <p>
+   * protected String serviceId;
+   * <p>
+   * protected DiscoveryClient discoveryClient;
+   * <p>
+   * protected String getRandomServiceInstance() {
+   * List<ServiceInstance> serviceInstances = discoveryClient.getInstances(serviceId);
+   * Random random = new Random();
+   * return serviceInstances.get(random.nextInt(serviceInstances.size())).getUri().toString();
+   * }
+   *
+   * @param urlResolver of the Camunda BPM Platform REST API
+   * @return the builder
+   * @Override public String getBaseUrl() {
+   * return getRandomServiceInstance();
+   * }
+   * }
+   */
```

Comment:
🔧 I think code should be formatted differently and `@param`/`@return` are misplaced at the moment.
```suggestion
  /**
   * Url resolver of the Camunda 7 REST API. This information is mandatory.
   * <p>
   * If the server is in a cluster or you are using spring cloud, you can create a class which implements UrlResolver..
   * <p>
   * this is a sample for spring cloud DiscoveryClient
   * <pre>
   * {@code
   * public class CustomUrlResolver implements UrlResolver{
   * protected String serviceId;
   *
   * protected DiscoveryClient discoveryClient;
   *
   *   protected String getRandomServiceInstance() {
   *     List serviceInstances = discoveryClient.getInstances(serviceId);
   *     Random random = new Random();
   *
   *     return serviceInstances.get(random.nextInt(serviceInstances.size())).getUri().toString();
   *   }
   *
   *   public String getBaseUrl() {
   *     return getRandomServiceInstance();
   *   }
   * }
   * </pre>
   * @param urlResolver of the Camunda 7 REST API
   * @return the builder
   */
```